Initial Commit

asap opened the pull request.

asap committed:
Initial Commit of React Redux to get /r/aww

Simple app to pull from /r/aww, and render using react-redux.

I want to display the first X posts in /r/aww, filtering out NSFW
posts and non-image posts.

On load, I want the first image to be displayed as large as
possible, and the image list to be clickable. When clicked,
the selected image will become the main image.

howardjing commented on src/App.js:
```diff
@@ -0,0 +1,69 @@
+class App extends Component {
+  constructor(props) {
+    super(props)
+    this.getMoar = this.getMoar.bind(this)
```
we're using https://github.com/jeffmo/es-class-fields-and-static-properties (https://babeljs.io/docs/plugins/transform-class-properties/) so you can do

```
getMoar = () => { }
```

howardjing replied:
(not in this project obvi)

cogell commented on package.json:
```diff
@@ -0,0 +1,40 @@
+  "dependencies": {
+    "classnames": "^2.2.0",
+    "immutable": "^3.7.5",
+    "jquery": "^2.1.4",
```
yikes!

asap replied:
HAHA. I should probably remove that. I'm not using it. I honestly don't know where it came from.

cogell commented on src/App.js:
```diff
@@ -0,0 +1,69 @@
+  }
+}
+
+export default App
```
kill this line

cogell commented on src/Awwz.js:
```diff
@@ -0,0 +1,34 @@
+          awwz.map((aww, index) =>
+          <li key={index}>
+            <a href="#" onClick={ e => {
+                e.preventDefault()
```
move preventDefault into handleClick

cogell commented on src/Awwz.js:
```diff
@@ -0,0 +1,34 @@
+  }
+}
+
+Awwz.propTypes = {
```
:+1:

cogell commented on src/BigAww.js:
```diff
@@ -0,0 +1,18 @@
+import React, { Component, PropTypes } from 'react';
+
+class BigAww extends Component {
```
check out React pure functional components. This doesnt actually need to be a whole class. Mind blown when I learned about them

cogell commented on src/actions.js:
```diff
@@ -0,0 +1,107 @@
+ */
+
+function decodeHTMLEntities (str) {
+  // 'Borrowed' from http://stackoverflow.com/questions/5796718/html-entity-decode
```
:+1: nice citation

cogell commented on src/actions.js:
```diff
@@ -0,0 +1,107 @@
+}
+
+export function getMoreAwwz (awwz) {
+  console.warn("TBD - I haven't quite figured that part out yet.")
```
pagination is always a little annoying

cogell commented on package.json:
```diff
@@ -0,0 +1,40 @@
+  },
+  "dependencies": {
+    "classnames": "^2.2.0",
+    "immutable": "^3.7.5",
```
used?

asap replied:
No, copied from the dashboard package. Removing it.

cogell commented:
there some javascript style things I didnt comment on as this is a personal project -- I break my own rules when prototyping too.

howardjing commented on src/App.js:
```diff
@@ -0,0 +1,69 @@
+    return (
+      <div>
+        <BigAww aww={currentAww} />
+        {isFetching && awwz.length === 0 &&
```
I'm not a fan of 

```
{ cond1 && cond2 && <div>hey</div> }
```

Some alternatives I think are preferable are:

```
render () {
  let body;
  if (!isFetching) {
    if (awwz.length > 0) {
      body = <Awwz awwz={...} />;
    } else {
      body = <h2>NO CAN HAZ</h2>;
    }
  } else {
    body = <h2>IZ LOADING</h2>
  }

  return (
    <div>
      { body }
    </div>
  );
}
```

```
render () {
  return (
    <div>
      { this.renderBody(isFetching, awwz) }
    </div>
  );
}

renderBody (isFetching, awwz) {
  // ...
}
```

The second way is more in line with what we're doing elsewhere, see `renderSegmentInfo` for an example

howardjing commented on src/App.js:
```diff
@@ -0,0 +1,69 @@
+
+function mapStateToProps(state) {
+  const { awwz, currentAww } = state
+  const {
```
i think 

```
const isFetching = false;
```

is clearer

howardjing commented on src/Awwz.js:
```diff
@@ -0,0 +1,34 @@
+                e.preventDefault()
+                this.handleClick(aww)
+              }}>
+              <img src={aww.thumb} />
```
I think I'd prefer

```
let { handleClick } = this.props;
<img src={aww.thumb} onClick={handleClick.bind(this, aww)} />
```

howardjing replied:
getting rid of that `<a>` wrapper

howardjing commented on src/actions.js:
```diff
@@ -0,0 +1,107 @@
+}
+
+function receiveAwwz(awwz, json) {
+  let myAwwz = json.data.children.filter(child => IsValidAww(child.data))
```
i would prefer the `IsValidAww` to be `isValidAww`

asap committed:
Include CSS-Modules for styling

Using CSS Modules for component based styling.

http://glenmaddern.com/articles/css-modules

asap committed:
Update README.md

asap commented on README.md:
```diff
@@ -1,2 +1,5 @@
 My foray into React/Redux
+
+# Getting Started
+I haven't quite figured that part out yet...
```
Note to self: explains how to run this thing.

:disappointed:
